### PR TITLE
Share common code between PeekChar() and PeekChar(int)

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler, @alekseyts, please review.

**Comment by AlekseyTs** [comment]
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/7281/


Test Result (1 failure / +1)Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicLineCommit.CommitOnSave
--
  | Identified problems
  | CLR test failure (only matches the first, see "Latest Test Result" link for the rest)Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicLineCommit.CommitOnSave [FAIL]Indication 1

**Comment by AlekseyTs** [comment]
https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/7185/


There are errors when processing test resultsThere was an error processing the xunit test results: [xUnit] [ERROR] - The result file 'D:\j\workspace\windows_relea---8a39a417\Binaries\Release\UnitTests\VisualStudioIntegrationTests\xUnitResults\Roslyn.VisualStudio.IntegrationTests.dll.xml' for the metric 'xUnit.Net' is empty. The result file has been skipped.Indication 1
--
  | One or more test assemblies failed to runOne or more test assemblies did not produce results, possible causes: 1) The build was aborted out due to a long running or deadlocked test. 2) The test run for this assembly failed due to xUnit crashing (possibly due to a unhandled exception on a background thread or a stack overflow). 3) The test caused an assert dialog to be opened, which blocked the run. See the Console Log for more information. [xUnit] [ERROR] - The result file 'D:\j\workspace\windows_relea---8a39a417\Binaries\Release\UnitTests\VisualStudioIntegrationTests\xUnitResults\Roslyn.VisualStudio.IntegrationTests.dll.xml' for the metric 'xUnit.Net' is empty. The result file has been skipped.Indication 2

**Comment by AlekseyTs** [comment]
test windows_debug_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
test windows_release_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, we need a second sign-off for a small refactoring.

File: src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs [hunk 1]
```diff
@@ -317,17 +313,7 @@ public char PeekChar(int delta)
...
-                && !MoreChars())
-            {
-                ch = InvalidCharacter;
-            }
-            else
-            {
-                // N.B. MoreChars may update the offset.
-                ch = _characterWindow[_offset];
-            }
+            char ch = PeekChar();
```
Comment (agocke) [inline src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs]: I would be careful with this... `PeekChar(int)` is a very hot method in the compiler IIRC. I'm worried that calling `PeekChar()` isn't going to be inlined, causing significant overhead from the extra calls.

Comment (tannergooding) [inline src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs]: You could always move it to an intermediate (internal) method and put an `AgressiveInlining` hint on it. Wouldn't guarantee the inlining, but it should work for this case.

**Comment by cston** [comment]
Not needed.

**Pull request closed without merging**